react-ui/src/pages/submission/SubmissionForm.jsx

```diff
@@ -357,33 +356,45 @@ function SubmissionForm({
     }
   };
 
-  const disableSubmitting = submitting || isMutating;
+  const nameCharRemaining = CHAR_LIMIT_NAME - formState.name.value.length;
+  const descriptionCharRemaining = CHAR_LIMIT_DESCRIPTION - formState.description.value.length;
+  const disableEditing = submitting || isMutating;
+  const disableSubmitting = disableEditing || nameCharRemaining < 0 || descriptionCharRemaining < 0;
```

Comment:
Submit buttons should not be disabled for input errors for a couple of reasons. One is that this adds a spinner to the button in addition to disabling it, which doesn't convey that the user has made an input error. Even if we were to hide the spinner, the main problem with disabling a submission control is it is difficult for the user to figure out why the button is disabled unless an error message is displayed adjacent to the button. In this screenshot, I've scrolled to the bottom, but can't tell why the button is disabled because the field in question is above the fold: 
![image](https://github.com/heshammourad/vexillology-contests/assets/21203724/9a6aab47-abf1-42f7-b403-44f2522495f1)

If we implement it this way, the length checks should be added to `validateForm` and handled there as well. I'd recommend just putting a max length on the field and preventing entry beyond the limit, but even then we should probably add validation here in case someone uses the console to change the max length (or so something hacky like `onChange = (val) => val.substring(MAX_LENGTH);`). One of the benefits of `validateForm` is it will scroll to the first field that contains an error, so it's immediately clear what needs to be fixed.

Reply:
There's actually a built-in `maxLength` feature, so that was easy enough to add. I did add the form validation as well though.